server/src/main/java/org/elasticsearch/script/AbstractFieldScript.java

```diff
@@ -141,8 +141,20 @@ protected final void checkMaxSize(int currentSize) {
     public final void runForDoc(int docId) {
         prepareExecute();
         setDocument(docId);
-        execute();
+        try {
+            execute();
+        } catch (RuntimeException e) {
```

Comment:
should we be catching Exception or RuntimeException?

Reply:
I'm fine with both, will change if you think it should be broader.